lib/test_runner_test.go

```diff
@@ -54,34 +50,35 @@ func newTestRunner(t *testing.T) *testRunner {
 		t.Fatal(err)
 	}
 
-	bus := event.NewBus(ctx)
-
 	// A temporary directory for doing filesystem work.
 	tmpDir, err := ioutil.TempDir("", "lib_test_runner")
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	mr, err := testrepo.NewEmptyTestRepo(bus)
-	if err != nil {
-		t.Fatalf("error allocating test repo: %s", err.Error())
-	}
-	localResolver := dsref.SequentialResolver(mr.Dscache(), mr)
-	node, err := p2p.NewQriNode(mr, testcfg.DefaultP2PForTesting(), bus, localResolver)
-	if err != nil {
-		t.Fatal(err.Error())
+	qriPath := filepath.Join(tmpDir, "qri")
+	if err := os.Mkdir(qriPath, os.ModePerm); err != nil {
+		t.Fatal(err)
 	}
+	cfg := testcfg.DefaultMemConfigForTesting()
 
-	inst := NewInstanceFromConfigAndNodeAndBus(ctx, testcfg.DefaultConfigForTesting(), node, bus)
-	// Assign the repoPath, as some tests inspect it in order to save files in the repository.
-	// Not assigning a path will cause them to use the current directory, and then save
-	// into the sourcetree.
-	inst.repoPath = tmpDir
+	// ensure we create a mock registry client for testing
+	key := InstanceContextKey("RemoteClient")
+	ctx = context.WithValue(ctx, key, "mock")
```

Comment:
This is gross, and the only thing I think needs to be actively fixed. Perhaps an addition to the `config.Remotes` settings that allows you to explicitly use a "mock" client?

Reply:
yep I was thinking the same thing. I'd love an issue for this that we can fix in a follow up

Reply:
This code is my fault, and was added because I needed a way for tests in the `cmd` package (specifically pull_test.go) to use the MockRemoteClient. See: https://github.com/qri-io/qri/blob/05fc38a4cd33506695f28730d3f942c03f880cca/lib/lib.go#L517

It's not possible to inject dependencies into cobra commands, which is the start of this trouble. Adding an option to the configuration means then we need to document it and it becomes "public", which has other problems. What we really want is a way for tests that pull to be able to use this mockRemoteClient on all commands they execute. Maybe an environment variable is better?

Regardless, I'm all for coming up with a better solution than this hack.

Reply:
okay good to know, added your comment to the issue #1716